src/app.ts

```diff
@@ -76,10 +76,20 @@ if (fs.existsSync(igDataPath)) {
       'utf8'
     );
   }
+  for (const instance of outPackage.instances) {
+    const fileName = `${instance.resourceType}-${instance.id ?? instance.instanceName}`;
+    delete instance.instanceName; // Only needed for the file name - not a FHIR property
+    fs.writeFileSync(
+      path.join(program.out, `${fileName}.json`),
+      JSON.stringify(instance, null, 2),
+      'utf8'
+    );
```

Comment:
As @mint-thompson mentioned above, could it maybe be cleaner to make `InstanceDefinition` a class with a `toJSON` method that handled removing `instanceName` and a `getFileName` method? Then I think you could just make line `72` above `for (const sd of [...outPackage.profiles, ...outPackage.extensions, ...outPackage.instances])`, and it would work as is.

Reply:
I like this! I think at one point I had it implemented as a class, but I think I switched it back because it was so simple. I do like that it would simplify the output both here and in the IGExport and would resolve the other issue Mint pointed out where `instanceName` needs to be removed there. I'll update!